src/main/java/at/ac/fhcampuswien/fhmdb/service/MovieAPIService.java

```diff
@@ -0,0 +1,64 @@
+package at.ac.fhcampuswien.fhmdb.service;
+
+import at.ac.fhcampuswien.fhmdb.models.Movie;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MovieAPIService {
+
+    private static final String API = "https://prog2.fh-campuswien.ac.at/";
+
+    private MovieAPIService() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static List<Movie> getMovies() throws IOException {
+        Request request = new Request.Builder()
+                .url(API.concat("movies"))
+                .header("User-Agent", "http.agent")
+                .method("GET", null)
+                .build();
+
+        return makeMovieRequest(request);
+    }
+
+    public static List<Movie> getMoviesBy(String text, String genre, String releaseYear, String ratingFrom) throws IOException {
+        Request request = new Request.Builder()
+                .url(API.concat("movies?query=").concat(text)
+                        .concat("&genre=").concat(genre)
+                        .concat("&releaseYear=").concat(releaseYear)
+                        .concat("&ratingFrom=").concat(ratingFrom))
+                .header("User-Agent", "http.agent")
+                .method("GET", null)
+                .build();
```

Comment:
Instead of building the URL inside the request using `concat`, I would separate it, like so:

```
        HttpUrl requestURL = new HttpUrl.Builder()
                .scheme("https")
                .host("prog2.fh-campuswien.ac.at")
                .addPathSegment("movies")
                .addQueryParameter("genre", genre)
                .addQueryParameter("releaseYear", releaseYear)
                .addQueryParameter("ratingFrom", ratingFrom)
                .build();

        Request request = new Request.Builder()
                .url(requestURL)
                .header("User-Agent", "http.agent")
                .method("GET", null)
                .build();
```

The `scheme`, `host` and `pathSegment` can be saved in some constants